tests/language.test.ts

```diff
@@ -0,0 +1,107 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+import * as FlexWebChat from '@twilio/flex-webchat-ui';
+
+import { Configuration } from '../types';
+import { getChangeLanguageWebChat } from '../src/language';
+
+describe('getChangeLanguageWebChat', () => {
+  const mockManager: any = {
+    strings: {
+      greeting: 'Hello',
+      goodbye: 'Goodbye',
+    },
+    store: {
+      getState: jest.fn().mockReturnValue({ flex: { config: { language: 'en-US' } } }),
+    },
+    updateConfig: jest.fn(),
+  };
+
+  const mockConfig: Configuration = {
+    accountSid: 'test1',
+    flexFlowSid: 'flow1',
+    defaultLanguage: 'en-US',
+    translations: {
+      'en-US': {
+        greeting: 'Hello',
+        goodbye: 'Goodbye',
+      },
+      'fr-CA': {
+        greeting: 'Bonjour',
+        goodbye: 'Au revoir',
+      },
+    },
+    preEngagementConfig: {
+      description: 'Please fill out the form below',
+      submitLabel: 'Submit',
+      fields: [
+        {
+          type: 'InputItem',
+          label: 'Name',
+          attributes: { name: 'name', placeholder: 'Enter your name' },
+        },
+        {
+          type: 'SelectItem',
+          label: 'Language',
+          attributes: { name: 'language' },
+          options: [
+            { label: 'English', value: 'en-US' },
+            { label: 'French', value: 'fr-CA' },
+          ],
+        },
+      ],
+    },
+    mapHelplineLanguage(helpline: string): string {
+      throw new Error('Function not implemented.');
+    },
+    captureIp: false,
+    contactType: 'ip',
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should set the default language when no language is provided', () => {
+    const changeLanguage = getChangeLanguageWebChat(mockManager, mockConfig);
+    changeLanguage('');
+
+    expect(mockManager.store.getState().flex.config.language).toBe('en-US');
+    expect(mockManager.strings.greeting).toBe('Hello');
+    expect(mockManager.strings.goodbye).toBe('Goodbye');
+    expect(mockManager.updateConfig).toHaveBeenCalledWith({ preEngagementConfig: mockConfig.preEngagementConfig });
+  });
+
+  it('should set a new language when a language is provided', () => {
+    const changeLanguage = getChangeLanguageWebChat(mockManager, mockConfig);
+    changeLanguage('fr-CA');
+
+    expect(mockManager.store.getState().flex.config.language).toBe('fr-CA');
+    expect(mockManager.strings.greeting).toBe('Bonjour');
+    expect(mockManager.strings.goodbye).toBe('Au revoir');
+    expect(mockManager.updateConfig).toHaveBeenCalledWith({ preEngagementConfig: expect.any(Object) });
+
+    const updatedPreEngagementConfig = mockManager.updateConfig.mock.calls[0][0].preEngagementConfig;
+    expect(updatedPreEngagementConfig.description).toBe('Please fill out the form below');
```

Comment:
You can use expect(updatedPreEngagementConfig).toMatchObject({ ... }) if you want to check multiple parameters on an object at once - it will do a partial match, verify all the properties you specify exist on the object with the correct value, but not fail if extra properties also exist on the object